catalog/core/catalog-core-validator/src/main/java/ddf/catalog/validation/impl/ValidationParser.java

```diff
@@ -0,0 +1,266 @@
+                    entry.getValue().stored,
+                    entry.getValue().tokenized,
+                    entry.getValue().multivalued,
+                    BasicTypes.getAttributeType(entry.getValue().type));
+            attributeRegistry.registerAttribute(descriptor);
+
+        }
+    }
+
+    private void parseMetacardTypes(List<Outer.MetacardType> metacardTypes) {
+        BundleContext context = FrameworkUtil.getBundle(this.getClass())
+                .getBundleContext();
+        for (Outer.MetacardType metacardType : metacardTypes) {
+            Set<AttributeDescriptor> attributeDescriptors = new HashSet<>();
+            Set<String> requiredAttributes = new HashSet<>();
+
+            for (Map.Entry<String, Outer.MetacardAttribute> entry : metacardType.attributes.entrySet()) {
+                attributeDescriptors.add(attributeRegistry.getAttributeDescriptor(entry.getKey()));
+                if (entry.getValue().required) {
+                    requiredAttributes.add(entry.getKey());
+                }
+            }
+            if (!requiredAttributes.isEmpty()) {
+                ReportingMetacardValidator validator = new RequiredAttributesMetacardValidator(
+                        metacardType.type,
+                        requiredAttributes);
+                // TODO (RCZ) - do we need any service properties?
+                context.registerService(ReportingMetacardValidator.class, validator, null);
+
+            }
+            // TODO (RCZ) - What properties do we want?
+            Dictionary<String, ?> properties = new Hashtable<>();
+            MetacardType type = new MetacardTypeImpl(metacardType.type, attributeDescriptors);
+            context.registerService(MetacardType.class, type, properties);
+
+        }
+    }
+
+    private void parseValidators(Map<String, List<Outer.Validator>> validators) {
+        for (Map.Entry<String, List<Outer.Validator>> entry : validators.entrySet()) {
+            List<AttributeValidator> attributeValidators = validatorFactory(entry.getValue());
+            attributeValidatorRegistry.registerValidators(entry.getKey(), attributeValidators);
+        }
+    }
+
+    private List<AttributeValidator> validatorFactory(List<Outer.Validator> validators) {
+        return validators.stream()
+                .filter(Objects::nonNull)
+                .filter(v -> StringUtils.isNotBlank(v.validator))
+                .map(this::getValidator)
+                .collect(Collectors.toList());
+    }
+
+    private AttributeValidator getValidator(Outer.Validator validator) {
+        switch (validator.validator) {
+        case "size":
+            long lmin = Long.parseLong(validator.arguments.get(0));
+            long lmax = Long.parseLong(validator.arguments.get(1));
+            return new SizeValidator(lmin, lmax);
+
+        case "pattern":
+            String regex = validator.arguments.get(0);
+            return new PatternValidator(regex);
+
+        case "pastdate":
```

Comment:
Are those strings used anywhere else?

On a related note, I dislike switches over a parameter value whose cases are not constants. Admittedly I am lacking some context of how this class is used, but I would have expected these cases to be `public static final` constants somewhere, with javadoc.